packages/next/src/lib/generate-interception-routes-rewrites.ts

```diff
@@ -0,0 +1,53 @@
+import { pathToRegexp } from 'next/dist/compiled/path-to-regexp'
+import { NEXT_URL } from '../client/components/app-router-headers'
+import {
+  extractInterceptionRouteInformation,
+  isInterceptionRouteAppPath,
+} from '../server/future/helpers/interception-routes'
+import { Rewrite } from './load-custom-routes'
+
+// a function that converts normalised paths (e.g. /foo/[bar]/[baz]) to the format expected by pathToRegexp (e.g. /foo/:bar/:baz)
+function toPathToRegexpPath(path: string): string {
+  return path.replace(/\[([^\]]+)\]/g, ':$1')
+}
+
+export function generateInterceptionRoutesRewrites(
+  appPaths: string[]
+): Rewrite[] {
+  const rewrites: Rewrite[] = []
+
+  for (const appPath of appPaths) {
```

Comment:
We need to make this route more generic so that we can avoiding adding one route for every intercepting path otherwise it's gonna hit route limits currently. 

Can we make this more generic so that it can match `src: '/:path((?!_next).*)'` and `has: { type: 'header', key: 'RSC-Intercepted-Path', value: '(?<interceptedPath>.*)' }` then `dest: '/:interceptedPath*'`